frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -117,8 +118,15 @@ Visitor::profile_t TypeInference::init_apply(const IR::Node* node) {
 }
 
 void TypeInference::end_apply(const IR::Node* node) {
-    if (readOnly && !(*node == *initialNode))
-        BUG("%1%: typechecker mutated node %2%", dbp(node), dbp(initialNode));
+    if (readOnly && !(*node == *initialNode)) {
+        ToP4 top4(&std::cout, true, nullptr);
+        std::cout << "Initial program" << std::endl;
```

Comment:
Yes, this block ends with a BUG call, so all these statements just print more information to help analyze the bug.